benchopt/tests/__init__.py

```diff
@@ -12,9 +12,9 @@
 REQUIREMENT_BENCHMARK_PATH = TEST_BENCHMARK_DIR / 'requirement_benchmark'
 
 # Pattern to select specific datasets or solvers.
-SELECT_ONE_SIMULATED = r'simulated*200*rho=0]'
-SELECT_ONE_PGD = r'python-pgd*step_size=1]'
-SELECT_ONE_OBJECTIVE = r'dummy*reg=0.1]'
+SELECT_ONE_SIMULATED = r'simulated[n_features=200,rho=0]'
+SELECT_ONE_PGD = r'python-pgd[step_size=1]'
+SELECT_ONE_OBJECTIVE = r'dummy[reg=0.1]'
```

Comment:
No it does not work anymore.
We need a convention to isolate the parameters from the rest (I made the assumption that parameters are within "[]"). Also, I am not sure it makes sense to have a regex on the parameters.

The previous filters would need to be changed into:
- `'python-pgd*step_size=1]'`  --> `'python-pgd*[step_size=1]'`
- `'dummy*reg=0.1]'`  --> `'dummy*[reg=0.1]'` 
- `'simulated*200*rho=0]'` --> `'simulated*[n_features=200,rho=0]'`

Reply:
I agree with you that it does not make sense to have `*` in the args anymore and that we should raise error on incomplete `[/]` pairs. I also agree for no-regexp in the parameters.

We should add `$` in the `python-pgd` pattern by default, and make it possible to use `python-pgd*[step_size=1]` to select both `python-pgd` and `python-pgd-with_cb`, that way, this will be less surprising to users as using full name should not trigger other ones.

Reply:
Well, it would be weird to have at the same time :
- `python-pgd` select `python-pgd` and not `python-pgd-with-cb`
- `dummy` select `Dummy Sparse Regression`

For now, selecting the solver/dataset/objective supports regex, so the user need to explicitly add "$" (`python-pgd$`) to get only `python-pgd`.

Reply:
Makes sense. See bd979bc3bc53af65a81fa86d660a5e159ec59408.